tools/src/test/java/org/apache/kafka/tools/consumer/ConsoleConsumerTest.java

```diff
@@ -396,42 +396,22 @@ public void testGroupMetadataMessageFormatter(ClusterInstance cluster) throws Ex
-                    JsonNode keyNode = jsonNode.get("key");
-                    GroupMetadataKey groupMetadataKey =
-                        GroupMetadataKeyJsonConverter.read(keyNode.get("data"), GroupMetadataKey.HIGHEST_SUPPORTED_VERSION);
-                    assertNotNull(groupMetadataKey);
-                    assertEquals(groupId, groupMetadataKey.group());
-
-                    JsonNode valueNode = jsonNode.get("value");
-                    GroupMetadataValue groupMetadataValue =
-                        GroupMetadataValueJsonConverter.read(valueNode.get("data"), GroupMetadataValue.HIGHEST_SUPPORTED_VERSION);
-                    assertNotNull(groupMetadataValue);
-                    assertEquals("", groupMetadataValue.protocolType());
-                    assertEquals(0, groupMetadataValue.generation());
-                    assertNull(groupMetadataValue.protocol());
-                    assertNull(groupMetadataValue.leader());
-                    assertEquals(0, groupMetadataValue.members().size());
-                } else {
-                    JsonNode keyNode = jsonNode.get("key");
-                    GroupMetadataKey groupMetadataKey =
-                        GroupMetadataKeyJsonConverter.read(keyNode.get("data"), GroupMetadataKey.HIGHEST_SUPPORTED_VERSION);
-                    assertNotNull(groupMetadataKey);
-                    assertEquals(groupId, groupMetadataKey.group());
-
-                    JsonNode valueNode = jsonNode.get("value");
-                    GroupMetadataValue groupMetadataValue =
-                        GroupMetadataValueJsonConverter.read(valueNode.get("data"), GroupMetadataValue.HIGHEST_SUPPORTED_VERSION);
-                    assertNotNull(groupMetadataValue);
-                    assertEquals("consumer", groupMetadataValue.protocolType());
-                    assertEquals(1, groupMetadataValue.generation());
-                    assertEquals("range", groupMetadataValue.protocol());
-                    assertNotNull(groupMetadataValue.leader());
-                    assertEquals(1, groupMetadataValue.members().size());
-                }
+                // The group coordinator writes an empty group metadata record when the group is created for the first time
+                JsonNode keyNode = jsonNode.get("key");
+                GroupMetadataKey groupMetadataKey =
+                    GroupMetadataKeyJsonConverter.read(keyNode.get("data"), GroupMetadataKey.HIGHEST_SUPPORTED_VERSION);
+                assertNotNull(groupMetadataKey);
+                assertEquals(groupId, groupMetadataKey.group());
+
+                JsonNode valueNode = jsonNode.get("value");
+                GroupMetadataValue groupMetadataValue =
+                    GroupMetadataValueJsonConverter.read(valueNode.get("data"), GroupMetadataValue.HIGHEST_SUPPORTED_VERSION);
+                assertNotNull(groupMetadataValue);
+                assertEquals("", groupMetadataValue.protocolType());
```

Comment:
nit: `assertTrue(groupMetadataValue.protocolType().isEmpty());`

Reply:
Since the aim of this patch is to remove the code, I prefer to refine this type of change in a follow-up if needed.